website/projects/awssync.py

```diff
@@ -36,3 +219,499 @@ def create_aws_organization(self):
+            self.logger.error("Something went wrong attaching an SCP policy to a target.")
+            self.logger.debug(f"{error}")
+            self.logger.debug(f"{error.response}")
+
+    def check_aws_api_connection(self):
+        """
+        Check whether boto3 can connect to AWS API with current credentials.
+
+        :returns: First tuple element always exists and indicates success.
+                  Second tuple element is contains information about the entity
+                  who made the successful API call and None otherwise.
+        """
+        client_sts = boto3.client("sts")
+        try:
+            caller_identity_info = client_sts.get_caller_identity()
+        except (NoCredentialsError, ClientError) as error:
+            self.logger.info("Establishing AWS API connection failed.")
+            self.logger.debug(error)
+            return False, None
+        else:
+            self.logger.info("Establishing AWS API connection succeeded.")
+
+        return True, caller_identity_info
+
+    def check_iam_policy(self, iam_user_arn, desired_actions):
+        """
+        Check for the specified IAM user ARN whether the actions in list \
+        desired_actions are allowed according to its IAM policy.
+
+        :param iam_user_arn: ARN of the IAM user being checked.
+        :param iam_actions: List of AWS API actions to check.
+        :returns: True iff all actions in desired_actions are allowed.
+        """
+        client_iam = boto3.client("iam")
+
+        try:
+            response = client_iam.simulate_principal_policy(PolicySourceArn=iam_user_arn, ActionNames=desired_actions)
+        except ClientError as error:
+            self.logger.info("AWS API actions check failed.")
+            self.logger.debug(error)
+            return False
+
+        success = True
+        for evaluation_result in response["EvaluationResults"]:
+            action_name = evaluation_result["EvalActionName"]
+            if evaluation_result["EvalDecision"] != "allowed":
+                self.logger.debug(f"The AWS API action {action_name} is denied for IAM user {iam_user_arn}.")
+                success = False
+
+        if success:
+            self.logger.info("AWS API actions check succeeded.")
+
+        return success
+
+    def check_organization_existence(self):
+        """
+        Check whether an AWS organization exists for the AWS API caller's account.
+
+        :returns: First tuple element always exists and indicates success.
+                  Second tuple element is describes properties of the organization and None otherwise.
+        """
+        client_organizations = boto3.client("organizations")
+
+        try:
+            response_org = client_organizations.describe_organization()
+        except ClientError as error:
+            self.logger.info("AWS organization existence check failed.")
+            self.logger.debug(error)
+            return False, None
+        else:
```

Comment:
Again, else here is not necessary when you return false and none in the exception